skiko/src/jvmMain/kotlin/org/jetbrains/skiko/redrawer/LinuxOpenGLRedrawer.kt

```diff
@@ -4,28 +4,40 @@ import kotlinx.coroutines.*
 import kotlinx.coroutines.swing.Swing
 import org.jetbrains.skiko.*
 import org.jetbrains.skiko.context.OpenGLContextHandler
+import java.util.concurrent.Executors
 
 internal class LinuxOpenGLRedrawer(
     private val layer: SkiaLayer,
     private val properties: SkiaLayerProperties
 ) : Redrawer {
     private var isDisposed = false
+
     private var context = 0L
-    private val swapInterval = if (properties.isVsyncEnabled) 1 else 0
+    private val defaultSwapInterval = if (properties.isVsyncEnabled) 1 else 0
+    private var swapInterval = -1
+
+    private fun LinuxDrawingSurface.setSwapIntervalFast(swapInterval: Int) {
+        if (this@LinuxOpenGLRedrawer.swapInterval != swapInterval) {
+            setSwapInterval(swapInterval)
+        }
+    }
 
     init {
-    	layer.backedLayer.lockLinuxDrawingSurface {
-            context = it.createContext(layer.transparency)
-            if (context == 0L) {
-                throw RenderException("Cannot create Linux GL context")
-            }
-            it.makeCurrent(context)
-            if (!isVideoCardSupported(layer.renderApi)) {
-                throw RenderException("Cannot create Linux GL context")
+        runBlocking {
+            inDrawThread {
+                context = it.createContext(layer.transparency)
+                if (context == 0L) {
+                    throw RenderException("Cannot create Linux GL context")
+                }
+                it.makeCurrent(context)
+                if (!isVideoCardSupported(layer.renderApi)) {
+                    throw RenderException("Cannot create Linux GL context")
+                }
+                it.setSwapIntervalFast(defaultSwapInterval)
```

Comment:
Seems here `setSwapInterval` shall be used.

Reply:
`setSwapIntervalFast` implemented in a way, that it has uninitialized value `-1`, so setSwapIntervalFast will work, and it is more consistent with the other code than simple setSwapInterval